cni/pkg/nodeagent/net_linux.go

```diff
@@ -0,0 +1,300 @@
+//go:build linux
+// +build linux
+
```

Comment:
In general for all of these I don't think we need _both_ the directives and the `_linux.go`/`_linux_test.go` filename suffixes, and if we have to pick one to lint for as a repo convention, I'd prefer to use the file naming convention versus the directives.

Picking one or the other kinda forces consistency - using both means it's easier for people to use one or the other method interchangeably and slip thru CI.

(unless there's some Go standards doc I missed that recommends using both at the same time)

Reply:
Ah, I thought the suffix was just convention and not enforced by the compiler; TIL. I also prefer the suffixes for readability purposes. That being said, I do think once we add windows support, we'll need more complex directives for unspecified, e.g. `go:build !linux !windows`

Reply:
I _think_ you do `x_linux`, `x_windows`, `x_unspecified` maybe. Not sure

Reply:
I think the way to handle this is to just not have `x_unspecified.go` at all, it doesn't mean anything anyway.

(if we are worried about being building on macOS, we can have `x_unix.go` instead of `unspecified` and call it a day)

Reply:
Yeah you basically then have to have files for every OS you might try to build on. `x_unix.go` doesn't work FYI: https://groups.google.com/g/golang-nuts/c/_onLLh3RUO8/m/UIt5XeZIAgAJ. You need build constraints there

Reply:
> files for every OS you might try to build on

Yeah that's a max of three for us 

- Linux
- Windows
- $anything-else (really just `darwin`)

Renaming all the `_unspecified.go` stuff to `_darwin.go` should be effectively equivalent.